src/PaperG/FirehoundBlob/Facebook/FacebookCreative.php

```diff
@@ -58,7 +63,9 @@ public function getObjects()
      */
     public function setType($type)
     {
-        $this->type = $type;
+        if (in_array($type, self::$validAdTypes)) {
+            $this->type = $type;    
+        }
```

Comment:
Thinking more about this, I think you're right. Quietly ignoring the value could seem like a recipe for disaster in the long run. 

My current thought is to allow all values, and then in the validator spit out what the incorrect value is. 

Another thought that came up while writing this is giving the validators the ability to coerce  values, which stemmed from this: http://json-schema.org/ I think writing the schema once and using that in code is very powerful and enables us to cut down friction between design and implementation. 

A lot of thoughts in my head for the validator stuff, yet all that's needed for this PR _for now_ is allow all values and then check the type on `isValid()`.